dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/utils/HadoopUtils.java

```diff
@@ -90,16 +90,27 @@ public HadoopUtils load(String key) throws Exception {
     private Configuration configuration;
     private FileSystem fs;
 
-    private HadoopUtils() {
+    private HadoopUtils(boolean initial) {
+        if (!initial) {
+            return;
+        }
         init();
         initHdfsPath();
     }
 
+    private HadoopUtils() {
+        this(true);
+    }
+
     public static HadoopUtils getInstance() {
 
         return cache.getUnchecked(HADOOP_UTILS_KEY);
     }
 
+    public static HadoopUtils getInstanceForTest() {
+        return new HadoopUtils(false);
+    }
+
```

Comment:
This approach is not recommended. just for the test.

Reply:
So directly using `new HadoopUtils(false)` ? Or other approach ?

Reply:
I prefer other ways. Or it is recommended to refactor it, and it is not recommended to use static code blocks in this case.

Reply:
Could you help give me some suggestions? Have no ideas on it. Thanks~

Reply:
An important factor in unit testing is structuring your code so that it is suitable for testing. code with excessive static initialization is not a structure that's easily tested. Even you can see that the famous Mockito does not support static method testing so far, my suggestion is You refactor the code and give up the use of static methods. this scenario is not applicable. we cannot use mocks to complete the mocks of external components.